src/node_modules/_file-explorer.js

```diff
@@ -0,0 +1,292 @@
+var setval = require('setval')
+var yo = require('yo-yo')
+var csjs = require('csjs-inject')
+var Treeview = require('ethereum-remix').ui.TreeView
+
+var css = csjs`
+  .treeview {
+    display: block;
+    margin-top: 7px;
+    width: 20%;
+  }
+  .folder,
+  .file {
+    font-size: 14px;
+  }
+  .hasFocus {
+    background-color: #F4F6FF;
+  }
+  .rename {
+    background-color: white;
+  }
+  .remove,
+  .save {
+    align-self: center;
+    margin-right: 3%;
+  }
+  .activeMode {
+    display: flex;
+    justify-content: space-between;
+  }
+`
+module.exports = fileExplorer
+
+function fileExplorer (container, files, editor) {
```

Comment:
We are currently trying to disentangle the various modules in a way that modules do not know anything about other modules. Each module only registers for events created by other modules or gets an object with callback functions and `app.js` connects the modules with each other. In this case, the file explorer would register on events by the `files` module (it only gets the event machine of files, not the files object itself) and it would receive an object with callbacks from `app.js` of the form `{ renameFile: function(oldName, newName) { /* call the respective function of the "files" module */ }, renameDirectory: function(...) { ... }, deleteFile: function(...) { ... }`. This way, the tree view and its action logic are cleanly separated.